src/Analyzers/CSharp/CodeFixes/UseParameterNullChecking/CSharpUseParameterNullCheckingCodeFixProvider.cs

```diff
@@ -0,0 +1,109 @@
+                        {
+                            BinaryExpressionSyntax binary => (binary.Left, binary.Right),
+                            IsPatternExpressionSyntax isPattern => (isPattern.Expression, ((ConstantPatternSyntax)isPattern.Pattern).Expression),
+                            InvocationExpressionSyntax { ArgumentList.Arguments: var arguments } => (arguments[0].Expression, arguments[1].Expression),
+                            _ => throw ExceptionUtilities.UnexpectedValue(ifStatement.Kind())
+                        };
+
+                        // one of the sides of the binary must be a parameter
+                        var parameterInIf = (IParameterSymbol?)model.GetSymbolInfo(unwrapCast(left), cancellationToken).Symbol
+                            ?? (IParameterSymbol)model.GetSymbolInfo(unwrapCast(right), cancellationToken).Symbol!;
+
+                        var parameterSyntax = (ParameterSyntax)parameterInIf.DeclaringSyntaxReferences[0].GetSyntax(cancellationToken);
+                        editor.RemoveNode(ifStatement);
+                        editor.ReplaceNode(parameterSyntax, parameterSyntax.WithExclamationExclamationToken(SyntaxFactory.Token(SyntaxKind.ExclamationExclamationToken)));
+                        break;
+                    case ExpressionStatementSyntax expressionStatement:
+                        // this.item = item ?? throw new ArgumentNullException(nameof(item));
+                        var assignment = (AssignmentExpressionSyntax)expressionStatement.Expression;
+                        var nullCoalescing = (BinaryExpressionSyntax)assignment.Right;
+                        var parameterReferenceSyntax = nullCoalescing.Left;
+
+                        var parameterSymbol = (IParameterSymbol)model.GetSymbolInfo(unwrapCast(parameterReferenceSyntax), cancellationToken).Symbol!;
+                        var parameterDeclarationSyntax = (ParameterSyntax)parameterSymbol.DeclaringSyntaxReferences[0].GetSyntax(cancellationToken);
+
+                        editor.ReplaceNode(nullCoalescing, parameterReferenceSyntax.WithAppendedTrailingTrivia(SyntaxFactory.ElasticMarker));
+                        editor.ReplaceNode(parameterDeclarationSyntax, parameterDeclarationSyntax.WithExclamationExclamationToken(SyntaxFactory.Token(SyntaxKind.ExclamationExclamationToken)));
+                        break;
+                }
+            }
+
+            static ExpressionSyntax unwrapCast(ExpressionSyntax expression)
+            {
+                if (expression is CastExpressionSyntax { Expression: var operand })
+                {
+                    return operand;
+                }
+
+                return expression;
+            }
+        }
+
+        private class MyCodeAction : CustomCodeActions.DocumentChangeAction
+        {
+            public MyCodeAction(string title, Func<CancellationToken, Task<Document>> createChangedDocument)
+                : base(title, createChangedDocument, title)
```

Comment:
equivalence key is recommended to be non-localized. The recommendation is to go for `nameof(CSharpAnalyzersResources.Use_parameter_null_checking)`